geoh5py/objects/text.py

```diff
@@ -0,0 +1,192 @@
+    @classmethod
+    def array_to_str(cls, float_list: np.ndarray | list):
+        if isinstance(float_list, np.ndarray | list):
+            float_list = "{" + ",".join([str(val) for val in float_list]) + "}"
+
+        return float_list
+
+
+class TextData(BaseModel):
+    """
+    Text mesh data.
+    """
+
+    text_data: list[TextEntry] = Field(
+        validation_alias=AliasChoices("text_data", "Text Data"),
+        serialization_alias="Text Data",
+    )
+
+
+class TextObject(Points):
+    """
+    Text object for annotated text labels in viewport.
+    """
+
+    _attribute_map: dict = Points._attribute_map.copy()
+    _attribute_map.update({"TextMesh Data": "text_mesh_data"})
+    _default_name = "Text Object"
+    _TYPE_UID: uuid.UUID = uuid.UUID("{c00905d1-bc3b-4d12-9f93-07fcf1450270}")
+
+    def __init__(
+        self,
+        vertices: np.ndarray | list | tuple | None = None,
+        text_mesh_data: str | None = None,
+        **kwargs,
+    ):
+        self._text_mesh_data: TextData
+
+        super().__init__(vertices=vertices, text_mesh_data=text_mesh_data, **kwargs)
+
+    @property
+    def text_mesh_data(self) -> TextData:
+        """
+        Text mesh data.
+        """
+
+        return self._text_mesh_data
+
+    @text_mesh_data.setter
+    def text_mesh_data(self, value: str | None):
+        """
+        Set the text mesh data.
+        """
+
+        self._text_mesh_data = self._validate_text_data(value)
+
+    @property
+    def extent(self):
+        """
+        Geography bounding box of the object.
+        """
+        return None
+
+    def mask_by_extent(self, extent: ndarray, inverse: bool = False) -> None:
+        """
+        Sub-class extension of :func:`~geoh5py.shared.entity.Entity.mask_by_extent`.
+        """
+        return None
+
+    def _validate_text_data(self, value: str | dict | None) -> TextData:
+        """
+        Validate the text data.
+        """
+        if value is None:
+            return TextData(
+                text_data=[
+                    TextEntry(text="<no-data>", starting_point=vert)
+                    for vert in self.vertices
+                ]
+            )
+
+        if isinstance(value, str):
+            value = loads(value)
+
+        if not isinstance(value, dict):
+            raise TypeError("The 'Text Data' must be a dictionary or a JSON string.")
+
+        mesh_data = TextData(**value)
+
+        if len(mesh_data.text_data) != self.n_vertices:
+            raise ValueError(
+                "The 'Text Data' dictionary must contain a list of len('n_vertices')."
+            )
+
+        return mesh_data
+
+    def _validate_value_length(self, values) -> list:
+        """
+        Validate the length of the values.
+        """
+        if not isinstance(values, (list, np.ndarray)):
+            values = [values] * self.n_vertices
+
+        if len(values) != self.n_vertices:
```

Comment:
Just an opened question:

why text data is dependant on points?
Couldn't be associated with another type of object?

Reply:
Points have the mechanics to deal with vertices, which this object also has.